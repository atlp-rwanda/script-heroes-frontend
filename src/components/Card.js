import React from 'react';
import PropTypes from 'prop-types';

const Card = ({property}) => {
    const {index,description, title, img } = property;
    return (
        <div id={`card-${index}`} className="card">
            <img src={img} alt={title} className='service-img' />
            <div className="details">
                <h1>{title}</h1>
                <p className="description">
                    {description}
                </p>
            </div>
        </div>
    )
}

Card.propTypes = {
    property: PropTypes.object.isRequired
}

export default Card;