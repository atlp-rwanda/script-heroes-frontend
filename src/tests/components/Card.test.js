import {shallow} from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'
import Card from '../../components/Card'

describe("Service card", ()=>{
    const property={
        _id: "0",
        index: 0,
        description: "This is the service",
        title: "Test Service",
        img: "imgsrc"
    }
    it("Should match snapshot", ()=>{
        const wrapper= shallow(<Card key={property._id} 
          property={property}/>)
        expect(toJSON(wrapper)).toMatchSnapshot()
    })   
})


