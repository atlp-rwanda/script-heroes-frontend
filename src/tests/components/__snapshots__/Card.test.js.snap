// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service card Should match snapshot 1`] = `
<div
  className="card"
  id="card-0"
>
  <img
    alt="Test Service"
    className="service-img"
    src="imgsrc"
  />
  <div
    className="details"
  >
    <h1
      className="card-title"
    >
      Test Service
    </h1>
    <p
      className="description"
    >
      This is the service
    </p>
  </div>
</div>
`;
